GOFevaluation/utils.py

```diff
@@ -47,6 +49,106 @@ def equiprobable_histogram(data_sample, reference_sample, n_partitions,
+        respectively. bin_edges[1] is a list of bin edges corresponding to the
+        partitions defined in bin_edges[0].
+    :rtype: list of arrays
+    """
+    xlim, ylim = get_plot_limits(data_sample)
+    be = []
+    if len(data_sample.shape) == 1:
+        bin_edges[0] = xlim[0]
+        bin_edges[-1] = xlim[1]
+        be = [bin_edges.copy(), None]
+    else:
+        be_first = bin_edges[0].copy()
+        be_second = bin_edges[1].copy()
+
+        if order == [0, 1]:
+            be_first[be_first == -np.inf] = xlim[0]
+            be_first[be_first == np.inf] = xlim[1]
+            be_second[be_second == -np.inf] = ylim[0]
+            be_second[be_second == np.inf] = ylim[1]
+        elif order == [1, 0]:
+            be_first[be_first == -np.inf] = ylim[0]
+            be_first[be_first == np.inf] = ylim[1]
+            be_second[be_second == -np.inf] = xlim[0]
+            be_second[be_second == np.inf] = xlim[1]
+        else:
+            raise ValueError(f'order {order} is not defined.')
+        be = [be_first, be_second]
+
+    return be
+
+
+def density_modifier(ns, be_first, be_second, data_sample):
+    """Creates limits for the edges of bins. Normally, bins
+    at the borders are infinite, this allows for all possible
+    datapoints to be binned. However, this makes it impossible
+    to calculate the areas of these border bins, and to show the
+    edges of these bins in a plot. This function uses the limits
+    and data_sample to limit the total area of all of the bins
+    so that they can be plotted and their area can be measured.
+
+    :param be_first: list of bin_edges in the first dimension,
+    :type be_first: array
+    :param be_first: list of bin_edges in the first dimension,
+    :type be_first: array
+    :param data_sample: Sample of unbinned data.
+    :type data_sample: array
+    :param order: Order in which the partitioning is performed,
+        defaults to None
+        [0, 1] : first bin x then bin y for each partition in x
+        [1, 0] : first bin y then bin x for each partition in y
+        if None, the natural order, i.e. [0, 1] is used. For 1D just put None.
+    :type order: list, optional
+    :return: Returns bin_edges.
+            1D: list of bin edges
+        2D: For order [0, 1]([1, 0]) these are the bin edges in x(y) and y(x)
+        respectively. bin_edges[1] is a list of bin edges corresponding to the
+        partitions defined in bin_edges[0].
+    :rtype: list of arrays
+    """
+    if len(data_sample.shape) == 1:
+        i = 0
+        for low, high in zip(be_first[:-1], be_first[1:]):
+            ns[i] = ns[i]/(high-low)
+            i += 1
+    else:
+        # plot rectangle for each bin
+        i = 0
+        for low_f, high_f in zip(be_first[:-1], be_first[1:]):
+            j = 0
+            for low_s, high_s in zip(be_second[i][:-1], be_second[i][1:]):
+                ns[i][j] = ns[i][j]/((high_f-low_f)*(high_s-low_s))
```

Comment:
This still has some pep8 clashes (We tuned the linter to be more strict about this in the future so we should also do it now, otherwise we have to reformat)